code/modules/materials/Mat_Fabrication.dm

```diff
@@ -334,6 +334,8 @@
 		W.set_loc(src)
 
 	attackby(var/obj/item/W , mob/user as mob)
+		if(istype(W, /obj/item/deconstructor))
+			return
```

Comment:
are you sure you want to `return` here and not `return ..()`?

Reply:
changed this- `deconstructor` uses `afterattack` for its logic but `atom/proc/attackby` has a nice visible message.